frb_codegen/src/lib.rs

```diff
@@ -40,10 +40,22 @@ mod parser;
 mod source_graph;
 mod target;
 mod transformer;
-mod utils;
+pub mod utils;
 use error::*;
 
 pub fn frb_codegen(config: &config::Opts, all_symbols: &[String]) -> anyhow::Result<()> {
+    frb_codegen_multi(config, &[config.clone()], 0, all_symbols)
+}
+
+/// the `all_configs` here is used only for multi-blocks
+pub fn frb_codegen_multi(
+    config: &config::Opts,
+    all_configs: &[config::Opts],
+    index: usize,
+    all_symbols: &[String],
+) -> anyhow::Result<()> {
+    assert_eq!(config.block_index, BlockIndex(index));
```

Comment:
If using the new API mentioned above, we can instead assert like:

```
assert!(configs.iter().enumerate().all(|index,config| config.index == BlockIndex(index))
```

then we are sure configs' indices are 0,1,2,...

Reply:
Refined in https://github.com/fzyzcjy/flutter_rust_bridge/pull/1024/commits/1217983475c2a047aa56593d4bb47da14ee354c1.